File: week_6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

def grade_level(text:str):
    letters = 0
    words = 1
    sentences = 0
    for c in text:
        if c == " ":
            words += 1
        elif c in ['?', '.', '!']:
            sentences += 1
        elif c.isalpha():
            letters += 1

    answer = calculate_index(letters, words, sentences)
    return answer

def calculate_index(count_letters: int, count_words: int, count_sentences:int):
    L = count_letters/count_words * 100.0
    S = count_sentences/count_words * 100.0
    result = 0.0588 * L - 0.296 * S - 15.8
    return round(result)

grade = grade_level(text)
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: week_2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int calculate_formula(int num_words, int num_sentences, int num_letters);
int grade_level(string text);
void return_range(int result);

int main(void)
{
    string text = get_string("Text: ");
    int grade = grade_level(text);
    return_range(grade);
}

int grade_level(string text)
{
    int word_count = 1;
    int sentence_count = 0;
    int letter_count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            word_count++;
        }
        else if (text[i] == '?' || text[i] == '.' || text[i] == '!')
        {
            sentence_count++;
        }
        else if (isalpha(text[i]))
        {
            letter_count++;
        }
    }
    int answer = calculate_formula(word_count, sentence_count, letter_count);
    return answer;
}

int calculate_formula(int num_words, int num_sentences, int num_letters)
{
    //average number of letters per 100 words in the text
    float L = (float) num_letters/(float) num_words * 100.0;
    //average number of sentences per 100 words in the text
    float S = (float) num_sentences/(float) num_words * 100.0;
    float result = 0.0588 * L - 0.296 * S - 15.8;
    return round(result);
}

void return_range(int result)
{
    if (result >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (result < 1)
    {
        printf ("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", result);
    }
}
